internal/provider/coredns/coredns.go

```diff
@@ -173,13 +153,13 @@ func (p *CoreDNSProvider) RecordsForHost(ctx context.Context, host string) ([]*e
 	var endpoints []*endpoint.Endpoint
 	// do our queries and gather up the answers
 	answers := map[string]map[string]*dns.Msg{}
-	p.logger.Info("RecordsForHost", "total nameservers", len(p.nameservers), "hosts", hosts)
+	p.logger.Info("recordsForHost", "total nameservers", len(p.nameservers), "hosts", hosts)
 	for _, nServer := range p.nameservers {
 		p.logger.Info("checking nameserver ", "address", *nServer)
 		nsAnswer, err := p.DNSQueryFunc(hosts, *nServer)
 		if err != nil {
 			//TODO prob need to handle dns errors better here
-			return endpoints, err
+			continue
```

Comment:
should we at least log here?

Reply:
Yeah I think we would expect each NS to be reachable and respond (eventually at least) If one of the NS is not returning results then you may have a functional record but one gateway is likely not receiving traffic

Reply:
Yeah, thinking on this, i think this wrong. I'm making this change to make testing easier which isn't really the correct solution.